source/Taboos/FiniteSubsetTaboo.lagda

```diff
@@ -0,0 +1,195 @@
+---
+title:        Kuratowski-finite Subset Taboo
+author:       Ayberk Tosun
+date-started: 2023-11-27
+---
+
+Based on a proof that I learned from Martín Escardó on 2023-11-21, though the
+proof I ended up writing here ended up being a bit different.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+open import UF.PropTrunc
+
+module Taboos.FiniteSubsetTaboo (pt : propositional-truncations-exist)
+                                (fe : Fun-Ext) where
+
+open import Fin.Kuratowski pt
+open import Fin.Type
+open import MLTT.Negation
+open import MLTT.Spartan
+open import Naturals.Order
+open import UF.DiscreteAndSeparated
+open import UF.ExcludedMiddle
+open import UF.ImageAndSurjection pt
+open import UF.Logic
+open import UF.Powerset
+open import UF.Powerset-Fin pt
+open import UF.Sets
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+open import UF.SubtypeClassifier-Properties using (Ω-is-set)
+
+open AllCombinators pt fe
+open PropositionalSubsetInclusionNotation fe
+open PropositionalTruncation pt
+
+\end{code}
+
+Let us first define a propositional version of `is-Kuratowski-finite-subset` for
+the sake of convenience.
+
+\begin{code}
+
+is-Kuratowski-finite-subsetₚ : {X : 𝓤  ̇} → 𝓟 X → Ω 𝓤
+is-Kuratowski-finite-subsetₚ P =
+ is-Kuratowski-finite-subset P , being-Kuratowski-finite-is-prop
+
+\end{code}
+
+We now define a predicate expressing the taboo we are interested in: given a
+type `X`, `Kuratowski-finiteness-is-hereditary X` expresses that for every
+Kuratowski-finite subset `F ⊆ X`, any further subset of `S ⊆ F` is also
+Kuratowski-finite.
+
+\begin{code}
+
+Kuratowski-finiteness-is-hereditary : 𝓤  ̇ → Ω (𝓤 ⁺)
+Kuratowski-finiteness-is-hereditary X =
+ Ɐ F ꞉ 𝓟 X , Ɐ S ꞉ 𝓟 X ,
+  S ⊆ₚ F ⇒ is-Kuratowski-finite-subsetₚ F ⇒ is-Kuratowski-finite-subsetₚ S
+
+\end{code}
+
+The result that we prove in this module is the following
+
+```
+  Kuratowski-finiteness-is-hereditary X → is-discrete X
+```
+
+We now prove two easy lemmas before we proceed to the proof of the result of
+interest.
+
+Lemma 1:
+
+\begin{code}
+
+having-empty-enumeration-entails-emptiness :
+ (X : 𝓤  ̇) → (e : 𝟘 {𝓤₀} → X) → is-surjection e → ¬ X
+having-empty-enumeration-entails-emptiness X e σ x =
+ ∥∥-rec 𝟘-is-prop (𝟘-elim ∘ pr₁) (σ x)
+
+\end{code}
+
+Lemma 2:
+
+\begin{code}
+
+having-nonempty-enumeration-entails-inhabitedness :
+ (X : 𝓤  ̇) (n : ℕ) → 0 <ℕ n → (e : Fin n → X) → is-surjection e → X
```

Comment:
Is there a reason for writing `0 <ℕ n` instead of just `0 < n`?

Reply:
Change as to use `_<_`. The reason was that I was not aware of the existence of the `Notation.Order` module. I discovered it yesterday, while doing something else.